messaging/src/main/java/org/axonframework/messaging/reactive/ReactorResultHandlerInterceptor.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2010-2020. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.messaging.reactive;
+
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.ResultMessage;
+import reactor.core.publisher.Flux;
+
+/**
+ * Interceptor that allows results to be intercepted and modified before they are handled. Implementations are required
+ * to operate on a {@link Flux} of results or return a new {@link Flux} which will be passed down the interceptor chain.
+ * Also, implementations may make decisions based on the message that was dispatched.
+ *
+ * @param <M> The type of the message for which the result is going to be intercepted
+ * @param <R> The type of the result to be intercepted
+ * @author Sara Pellegrini
+ * @since 4.4
+ */
+@FunctionalInterface
+public interface ReactorResultHandlerInterceptor<M extends Message<?>, R extends ResultMessage<?>> {
+
+    /**
+     * Intercepts result messages. It's possible to break the interceptor chain by returning {@link Flux#empty()} or
```

Comment:
Few ideas:

Historically in the regular interceptor, you have the chain control via` proceed() ` function (this is reactive way of doing it) 

Handling transient exception, via flat map you can convert error to empty publisher

This could be useful for subscription queries, with result interceptor you could define global rule when "infinitive" subscriber should stop listening for updates

You want to filter messages that fulfill client predicate (`filter` function can be used for this)

Reply:
Sweet, sounds good.
It would be _very_ helpful to add these suggestions to the Reference Guide (to build) around the Reactive Gateways, as this will definitely help out our users.